autoextract_spiders/middlewares.py

```diff
@@ -0,0 +1,91 @@
+"""
+Writes a single json file for each issued item in the target s3://bucket/folder.
+
+Usage:
+
+SPIDER_MIDDLEWARES = {
+    'autoextract_spiders.middlewares.S3ItemWriteMiddleware': 1,
+}
+
+To enable the middleware, set the target bucket:
+
+S3_WRITE_BUCKET = <bucket name>
+
+Spider that needs to write items to s3 must set following attributes:
+
+s3_write_folder - Folder where to write the output
+s3_write_item_type - Only items belonging to the given class are written to s3
+
+AWS credentials are taken from the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
+"""
+
+import os
+import json
+import logging
+from hashlib import md5
+from datetime import datetime
+
+from pytz import utc, timezone
+from s3fs import S3FileSystem
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+from scrapy.exporters import PythonItemExporter
+
+TIMEZONE = timezone('Europe/Stockholm')
```

Comment:
Please, don't hardcode this timezone, this is not suposed to be here.
We could make this an option per project, in the settings file, but the default should be UTC.